dev-docs/RFCs/v8.1/gpu-accelerated-polygon-clipping.md

```diff
@@ -0,0 +1,153 @@
+
+### Existing Research Work
+
+Finding a point inside a polygon or not is also referred as PIP test. There has been lot of research work recently, to accelerate the PIP test using GPU.
+
+
+Following is the list of papers I found and explored :
+
+1. Speeding up Large-Scale Point-in-Polygon Test Based Spatial Join on GPUs ([Paper](http://www-cs.ccny.cuny.edu/~jzhang/papers/bigspaital_cr.pdf))
+
+2. A cell-based point-in-polygon algorithm suitable for large sets of points ([Paper](https://www.sciencedirect.com/science/article/pii/S0098300401000371?via%3Dihub))
+
+3. GPU Rasterization for Real-Time Spatial Aggregation over Arbitrary Polygons ([Paper](http://www.vldb.org/pvldb/vol11/p352-zacharatou.pdf))
+
+4. A Simple and fast hardware-accelerated point-in-polygon test ([Paper](https://pdfs.semanticscholar.org/04a4/7b99d57cdf81bbd58a8dc21c2d15ac528855.pdf))
+
+All of above work uses more advanced Graphics APIs (such as OpenGL Compute Shaders, CUDA, SSBO, etc) and or not supported either by WebGL1 or WebGL2. This RFC proposes a method that can be implemented with in WebGL1 and WebGL2 limitations.
+
+
+## Proposed Approach
+
+A texture based filtering can be implemented to perform PIP test on GPU. The idea is, user provided polygon(s) is triangulated and rendered to an offline frame buffer. And then, given points are run through transform feedback loop, each point is translated into the texture space and above texture is sampled, result of this sample determines if the point is inside or outside the polygon.
+
+
+## Proposed New Components
+
+### `texturefilter` (ShaderModule)
+
+Provides shader functionality, for transforming input point to texture space and sample the texture to determine if the point is inside or outside the region defined by the texture. When performing polygon clipping, this texture is created by rendering polygons to it.
+
+- Uniforms:
+   - `bboxOrigin` (vec2)  // origin of the bounding box
+   - `bboxSize` (vec2)  // Sizes in X and Y directions of the bounding box
+   - `maskTexture` (sampler2D) // Texture created by rendering polygons
+- VS Function:
+     - `vec3 isWithInTextureMask(vec2 point)`: returns [flag, id, pointIndex]
+
+
+### GPUPolygonFilter (JS Class) (WebGL2 only)
+
+Class that builds a texture mask from given set of polygons, performs transform feedback loop on given set of points and generates the results. It internally uses `texturefilter` shader module.
+
+#### Methods
+
+##### Constructor
+
+Constructor has following arguments, it constructs required objects such as `Texture2D` and `Transform`.
+
+* `gl`(WebGL2Context): WebGL Context
+* `opts` (Optional, Default: {}): Object containing following optional fields.
```

Comment:
I'd just change this to take a single argument, an array of polygons, and get rid of the `id` property. The id can just be the array index. 

`new GPUPolygonFilter([[x1, y1], [x2, y2], ...]])`

Reply:
couple of reasons:
1. `polygon` format itself is array of arrays, for a complex polygon there is an extra array level. Making them an array might be very complex and confusing.
2. Explicit `id` gives more flelxibiliy if the user is activating few polygons at a time, from a given set. `id` don't have to change between invokes.